pkg/platform/runtime/runtime.go

```diff
@@ -20,15 +23,20 @@ type Runtime struct {
 	msgHandler  MessageHandler
 }
 
-func NewRuntime(commitID strfmt.UUID, owner string, projectName string, msgHandler MessageHandler) *Runtime {
-	var installPath string
-	if runtime.GOOS == "darwin" {
-		// mac doesn't use relocation so we can safely use a longer path
-		installPath = filepath.Join(config.CachePath(), owner, projectName)
-	} else {
-		installPath = filepath.Join(config.CachePath(), hash.ShortHash(owner, projectName))
+func NewRuntime(projectDir string, commitID strfmt.UUID, owner string, projectName string, msgHandler MessageHandler) (*Runtime, error) {
+	var resolvedProjectDir string
+	if projectDir != "" {
+		var err error
+		projectDir = strings.TrimSuffix(projectDir, constants.ConfigFileName)
+		resolvedProjectDir, err = fileutils.ResolveUniquePath(projectDir)
+		if err != nil {
+			return nil, errs.Wrap(err, "Failed to resolve unique file path to project dir")
```

Comment:
Seems like a good place to localize the error where we have some context about what actually went wrong.